### DISCUSSION TITLE
Slash command autocompete cache prevention

### DISCUSSION THREAD

**Discussion by XCanG** [Rapptz/discord.py#9880] [Q&A]
I didn't find proper information on this, but I have a case with a command that take 2 arguments: `name` and `date`.
Both parameters have autocomplete, however `date` choices is relative to selected `name`. So in date autocomplete I get `interaction.namespace.name` argument and based on it filter available dates for that name.

The problem goes with caching. For example I type command `/command [name: A] [date: 2]` and get some dates choices from that name that contain `2` in it, like `2024.07.06`
However if in that state I delete with backspace date variable and change name to `B`, then type `[date: 2]` again, the choices I get before would be restored and bot would not get hit with an autocomplete function call (I checked that with `print`).

So now I have obsolete values that somehow cached and wont update to new proper ones. Even if I delete whole command and re-type again it wont do that. Only until I restart bot. How to prevent this caching behavior?

**Comment by Rapptz** [comment]
This caching behaviour is from the client itself, it's not related to the library in any way. It's a long standing upstream issue: https://github.com/discord/discord-api-docs/issues/4804

As far as I know there's no workaround.

**Reply by XCanG** [reply]
I see, that's sad. But why does it reset on bot restart?

**Reply by Rapptz** [reply]
I'm not too sure about that, as far as I know the bot has no real control over this state since the caching is entirely client sided. Restarting the client should fix the cache issue.

**Reply by pythonmcpi** [reply]
IIRC the cache gets cleared if the user deletes their partially typed command (and retypes it). It might also get cleared when you switch channels.
